File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int left = 0;
        int n = heights.length;
        int right = n - 1;
        int maxAra = 0;
        int area = 0;

        while(left <= right){
            area = Math.min(heights[left],heights[right])*(right - left);
            maxAra = Math.max(maxAra,area);
            if(heights[left]<heights[right]){
                left++;
            }
            else{
                right--;
            }
        }

        return maxAra;
    }
}

File: Data Structures & Algorithms/max-area-of-island/submission-0.py
class Solution:
    def maxAreaOfIsland(self, grid: List[List[int]]) -> int:
        if not grid:
            return 0
    
        rows , cols = len(grid),len(grid[0])
        visit = set();
        area = 0;
        maxArea = 0;

        def bfs(r,c):
            q = collections.deque()
            q.append((r,c))
            visit.add((r,c))
            area = 1;
            directions = [[0,1],[0,-1],[1,0],[-1,0]]
        
            while q:
                row,col = q.popleft()

                for dr,dc in directions:
                    r = row + dr
                    c = col + dc

                    if(r in range(rows) and c in range(cols) and 
                    (r,c) not in visit 
                    and grid[r][c] == 1):
                        q.append((r,c))
                        visit.add((r,c))
                        area += 1
            
            return area


        for r in range(rows):
            for c in range(cols):
                if(grid[r][c] == 1 and (r,c) not in visit):
                    area = bfs(r,c)
                    maxArea = max(maxArea,area)
    

    

        return maxArea
        